xml/hwbp_registers.xml

```diff
@@ -741,15 +745,23 @@
             </value>
 
             <value v="1" name="before">
-            The trigger fired just before the instruction that triggered it was
-            retired, but after all preceding instructions are retired.
+            The trigger fired before the instruction that matched it was
+            retired, but after all preceding instructions are retired. This
+            explicitly allows for instructions partially executing.
             \Rxepc or \RcsrDpc (depending on \FcsrMcontrolSixAction) must be set
             to the virtual address of the instruction that matched.
 
-            If a load operation matched and \FcsrMcontrolSixSelect=1 then a
-            memory access has been performed (including any side effects of
-            performing such an access) even though the load has not updated its
-            destination register.
+            An instruction that caused a trigger to fire might be executed
+            partially. In that case not all memory accesses may have been
+            performed, and some registers may not have been updated. Executing
```

Comment:
Is this trying to say that ALL registers will not have been updated, or that some registers may have been updated?

Reply:
I'm trying to say that anything partial is OK, from 0% to 100%. E.g. if we had an instruction that pops 3 registers from the stack, any of those 3 registers (or none of them) may contain the new value. The important thing for the debugger is that nothing breaks if we now resume off that instruction by letting it execute again. (Can't account for non-idempotent memory, however.)

Reply:
@gfavor Is this OK, or would you like to see the language tweaked more?